server/block/smelter.go

```diff
@@ -0,0 +1,176 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/item/inventory"
+	"github.com/df-mc/dragonfly/server/world"
+	"math"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+// smelter is a struct that may be embedded by blocks that can smelt blocks and items, such as blast furnaces, furnaces,
+// and smokers.
+type smelter struct {
+	mu sync.Mutex
+
+	viewers   map[ContainerViewer]struct{}
+	inventory *inventory.Inventory
+
+	remainingDuration time.Duration
+	cookDuration      time.Duration
+	maxDuration       time.Duration
+	experience        int
+}
+
+// newSmelter initializes a new smelter with the given remaining, maximum, and cook durations and XP, and returns it.
+func newSmelter(remaining, max, cook time.Duration, experience int) *smelter {
+	s := &smelter{
+		viewers:           make(map[ContainerViewer]struct{}),
+		experience:        experience,
+		remainingDuration: remaining,
+		cookDuration:      cook,
+		maxDuration:       max,
+	}
+	s.inventory = inventory.New(3, func(slot int, item item.Stack) {
+		s.mu.Lock()
+		defer s.mu.Unlock()
+		for viewer := range s.viewers {
+			viewer.ViewSlotChange(slot, item)
+		}
+	})
+	return s
+}
+
+// Durations returns the remaining, maximum, and cook durations of the smelter.
+func (s *smelter) Durations() (time.Duration, time.Duration, time.Duration) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	return s.remainingDuration, s.maxDuration, s.cookDuration
+}
+
+// UpdateDurations updates the remaining, maximum, and cook durations of the smelter.
+func (s *smelter) UpdateDurations(remaining, max, cook time.Duration) {
```

Comment:
Is this method used anywhere? If not, just remove it.

Reply:
It now is used.